src/ds3.c

```diff
@@ -1166,6 +1193,87 @@ static ds3_object _parse_object(xmlDocPtr doc, xmlNodePtr contents_node) {
     return object;
 }
 
+static ds3_search_object _parse_search_object(xmlDocPtr doc, xmlNodePtr contents_node) {
+    xmlNodePtr child_node;
+    xmlChar* text;
+    ds3_search_object object;
```

Comment:
By allocating this on the stack and then returning it, this will lead to a condition called a 'dangling pointer.'  Use the `g_new0` call to allocate a pointer to memory which can then safely be returned.  `g_new0` also zero initializes the memory so you won't have to.